### ISSUE TITLE
Scalable hardpoint sprites

### ISSUE THREAD

**Issue by comnom** [endless-sky/endless-sky#3022]
Specifically in regards to fighters/drones and now turret mount sprites. We have the functionality for engine flares, but given the size disparity between ship classes I'd find it much easier to have scalable sprites rather than making sure the model can fit all fighters/turrets in the creation stage.

That is to say: By size disparity I mean that a tiny sparrow should be much smaller in game than a bactrian for instance, but due to gameplay concerns this isn't true (It's definitely smaller, but not as much as the mass would imply). Not that no effort is made, and not that I don't agree with the current state of affairs, just that reconciling the mass of the two ships alongside the size of the sprites means that the player (or camera) is much closer to the sparrow than the bactrian.

This has caused me a bit of issue in creating ships, as a few of the folks on discord know, and I feel it would ease the design and ship modelling process for newcomers if they could simply scale turret hardpoints and fighter/drone hardpoints in the ship definition rather than worry about it during the modelling phase.

On fighter hardpoint scaling: A super effort would involve using the same animation as ships landing on planets, but in reverse for launching.

Edit: Spelling

**Comment by tehhowch** [comment]
Can you provide a pseudo ship spec that uses this scaling? 

As I read your post, you are requesting some way to say `fighter x y 0.2 over` where the 0.2 means the carried sprite should be 20% of the normal in flight size - is that right?

**Comment by comnom** [comment]
Yea you got the scaling request right. 

I had some screenshots of ships that elyssaen sceenshotted for me with the turrets too big, but I lost them in a reinstall. Ill see if I can regrab them tomorrow, it was a korath ship and one of his Trin where it was just tough to scale the ship vs. the turret hardpoints.

**Comment by comnom** [comment]
I used the ship builder for images rather than a screenshot so that everything was bigger and I could slide them about. The worst offender on turret size is this guy with overlapping turrets: (the turret image used is 32x96 at2x, so smaller than a disruptor/slicer, but bigger than some others)

![metparoverlappingturrets](https://user-images.githubusercontent.com/22107649/30243386-ec8d8c6e-9575-11e7-89bf-3510baacab2c.PNG)

If we space things out we can kill the overlap:

![metparoverlappingturrets2](https://user-images.githubusercontent.com/22107649/30243387-029a569a-9576-11e7-97e9-65ab9c2301c0.PNG)

But even then that's a large chunk of the image taken up by turrets. A smaller scale turret would be preferable in my opinion. We'd also give newcomers the leeway to not have to know offhand that a slicer is the largest hardpoint sprite, so if they want no overlap they should test their scale with that, etc.

In short, all we'd need to do is model the ships, then tweak turrets and fighters after the fact, only having to worry about rough alignment for visible fighters and whatnot.

**Comment by KiLEdEnNis** [comment]
overlap wouldn't be that big of a problem if it's clear 1 turret is above another one, and to make sure we'd have to be able to specify that
but then when facing the other way it'd look plain weird
so I get why ya want scalable turrets

**Comment by comnom** [comment]
Yea, that's pretty much it for the turret hardpoints kile. Alot of my frustration in modelling ships post-hardpoint sprite comes from finding a flat spot for the turret. It's not so bad on heavy warships, because they're so big, but as you scale downwards it get tougher and tougher to fit a large flat area on things.

Fighters and drones are a slightly different beast, especially in higher tiers where its not uncommon to balance a carrier with 6 fighters or 12 drones. They take a huge amount of preallocated space on the model, and if you want them all visible your sprite would have to be massive. So I skip it usually. I show a pair, and that's that.

**Comment by endless-sky** [comment]
The turrets will only overlap if they happen to be pointed in certain directions.

I also think it'd look really weird if turrets are bigger on some ships than others. And for fighters, it would look even stranger because they would grow or shrink after being launched from the ship.

**Comment by comnom** [comment]
>The turrets will only overlap if they happen to be pointed in certain directions.

That's true, maybe I'm just being a bit anal about it.

>I also think it'd look really weird if turrets are bigger on some ships than others.
Not any weirder than the prerendered guns being different sizes to sell the scale better I would imagine. That was part of the idea really, that the relative size would better sell the scale like it does with engine flares.

Oh well though, just throwing it out there.

**Comment by KiLEdEnNis** [comment]
I think scaling turrets will be a good fix though, as long as we don't scale them to half their size, minor rescales could help

with the new turret hardpoint thing, a ship requires some good thought process on where to place turrets, which is kinda fun, the main problem is just existing ships

but I'll leave it at this then

**Comment by endless-sky** [comment]
I really don't see overlapping guns as a problem that needs to be "fixed." It's relatively common on real battleships to have multiple ranks of guns that overlap each other:

![hms_nelson_ 1931 _profile_drawing](https://user-images.githubusercontent.com/11450262/30249768-e606557e-9610-11e7-844e-e7e5d375eb43.png)

**Comment by KiLEdEnNis** [comment]
imagine those 3 at the front facing exactly the other way; the front 1 will have its barrels inside the second one, PLUS say they are all beam weapons, the front one would shoot a beam through the middle one

I don't really mind that much either though; one hardly notices at all unless they really focus on it
